pysoundcomparisons/mediacatalog.py

```diff
@@ -1,118 +1,32 @@
-from collections import OrderedDict
-from itertools import groupby
-from pathlib import Path
-import time
+from cdstarcat import Catalog
 
-from clldutils import jsonlib
-
-
-class MediaCatalog(object):
+class MediaCatalog(Catalog):
     __mimetypes = {
         'mp3': 'audio/mpeg',
         'ogg': 'audio/ogg',
         'wav': 'audio/wav',
     }
 
-    __uid_map = None
-
-    def __init__(self, repos):
-        self.path = Path(repos).joinpath('soundfiles', 'catalog.json')
-        if self.path.exists():
-            self.items = jsonlib.load(self.path, object_pairs_hook=OrderedDict)
-        else:
-            self.items = OrderedDict()
-
-    def __contains__(self, item):
-        return item in self.items
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        jsonlib.dump(self.items, self.path, indent=4)
-
-    def __getitem__(self, key):
-        return self.items[key]
-
-    def __len__(self):
-        return len(self.items)
+    __file_path_uid_map = None
 
-    def __iter__(self):
-        for key in self.items:
-            yield key
+    def file_path_uid_map(self):
```

Comment:
`clldutils.misc.lazyproperty` is exactly for this kind of thing, i.e. attributes which are expensive to compute and thus should be cached. So this would be
```python
@lazyproperty
def file_path_uid_map(self):
    return {obj.metadata['name'] : obj.id for obj in self}
```

Reply:
I looked for such an implementation - ok